src/main/java/de/tum/in/www1/exerciseapp/web/rest/QuizExerciseResource.java

```diff
@@ -77,16 +86,22 @@ public QuizExerciseResource(QuizExerciseRepository quizExerciseRepository,
             return ResponseEntity.badRequest().headers(HeaderUtil.createFailureAlert(ENTITY_NAME, "idexists", "A new quizExercise cannot already have an ID")).body(null);
```

Comment:
this method has become very complex. I suggest to move parts of it into a corresponding service and to extract multiple smaller methods out of it, each with a specific single purpose. Please apply the same idea to other complex methods in Rest Resources related to quizzes and statistics

Reply:
I created QuizExerciseService.java and moved some code to that.